### PR TITLE
ci + ios build

### PR DESCRIPTION
**Pull request by joshblum**
patch does the following:

- some CI related fixes (last failure is due to windows not yet being upgraded, i will likely merge over that and do another pass once it's up)
- fixes up calls to `runtime.GOOS` to match both `darwin` and `ios` (see https://tip.golang.org/doc/go1.16#darwin), using `libkb.RuntimeGroup()` where we can
- because of https://github.com/golang/go/issues/46860, currently requires a custom build of golang with this patch: https://go-review.googlesource.com/c/go/+/344401/9

### CODE REVIEW

File: go/libkb/home.go [hunk 1]
```diff
@@ -399,13 +401,13 @@ func (w Win32) MobileSharedHome(emptyOk bool) string {
...
+func NewHomeFinder(appName string, getHomeFromCmd ConfigGetter, getHomeFromConfig ConfigGetter, getMobileSharedHome ConfigGetter,
 	getRunMode RunModeGetter, getLog LogGetter, getenv EnvGetter) HomeFinder {
 	base := Base{appName, getHomeFromCmd, getHomeFromConfig, getMobileSharedHome, getRunMode, getLog, getenv}
-	switch osname {
-	case "windows":
+	switch RuntimeGroup() {
+	case keybase1.RuntimeGroup_WINDOWSLIKE:
 		return Win32{base}
-	case "darwin":
+	case keybase1.RuntimeGroup_DARWINLIKE:
```
Comment (joshblum) [inline go/libkb/home.go:410]: @mmaxim this was the breakage for the mobile build, we'd end up trying to configure ios with linux config paths

File: go/kbfs/redirector/main.go [hunk 2]
```diff
@@ -179,7 +179,7 @@ func (r *root) findKBFSMount(ctx context.Context) (
 		return "", err
 	}
 	fuseType := "fuse"
-	if runtime.GOOS == "darwin" {
+	if runtime.GOOS == "darwin" || runtime.GOOS == "ios" {
```
Comment (mmaxim) [inline go/kbfs/redirector/main.go:182]: Don't think we need any changes here, no redirector on iOS

File: go/libkb/constants.go [hunk 3]
```diff
@@ -98,18 +98,7 @@ var UserAgent = runtime.GOOS + ":" + "Keybase CLI (" + runtime.Version() + "):"
...
-		if isIOS {
-			os = "ios"
-		} else {
-			os = "mac"
-		}
-	} else {
-		os = runtime.GOOS
-	}
-
+	os := runtime.GOOS
```
Comment (mmaxim) [inline go/libkb/constants.go:101]: Is this the same? Do we want to preserve the "mac" behavior?

File: go/libkb/env.go [hunk 4]
```diff
@@ -844,7 +843,7 @@ func (e *Env) GetDisplayRawUntrustedOutput() bool {
 
 func (e *Env) GetAutoFork() bool {
 	// On !Darwin, we auto-fork by default
-	def := (runtime.GOOS != "darwin")
+	def := (RuntimeGroup() != keybase1.RuntimeGroup_DARWINLIKE)
```
Comment (mmaxim) [inline go/libkb/env.go:846]: This is irrelevant for iOS

**Review by mmaxim — APPROVED** [review]

File: go/libkb/constants.go (hunk 3 above)
Comment (joshblum) [reply go/libkb/constants.go:101]: we can, don't think it's required though

**Pull request merged**